Elm Support

andys8 opened the pull request:
This PR adds support for the [Elm programming language](https://elm-lang.org)

## Content

* Syntax
* Indentation
* Filetype

## History

The syntax defintion was originally created by Joseph Hager (https://github.com/ElmCast/elm-vim) but isn't maintained anymore. https://github.com/andys8/vim-elm-syntax is a fork which is now used by many users. This PR will add the syntax definition to Vim so users won't need a plugin for syntax anymore. Elm functionality is part of [elm-language-server](https://github.com/elm-tooling/elm-language-server).

andys8 committed:
Original elm syntax file

andys8 committed:
Original elm ftdetect and indent

andys8 committed:
Elm filetype

andys8 committed:
Elm add headers

brammool commented:
If I understand it correctly you didn't write this code.  Please add a copyright notice, make clear who is the author and what license applies to this code.
Ideally the Vim license can be used, but the original author must agree with that.

andys8 commented:
Thanks for the feedback!

This is the [original license](https://github.com/ElmCast/elm-vim/blob/master/LICENSE). I renamed the field "Previous Maintainer" to "Original Author", and added a "Copyright" and "License" field in the header. Does this match your expectations?

andys8 committed:
Elm: License and Copyright in Header

brammool commented:
> Thanks for the feedback!
> 
> This is the [original
> license](https://github.com/ElmCast/elm-vim/blob/master/LICENSE). I
> renamed the field "Previous Maintainer" to "Original Author", and
> added a "Copyright" and "License" field in the header. Does this match
> your expectations?

And the ftplugin you wrote yourself, right?
That should be OK then.

I'll make a patch for the filetype change, because it will also have a
test.  The rest will go out with the next runtime files update.

Please don't forget to send a new version when you make improvements.

-- 
hundred-and-one symptoms of being an internet addict:
211. Your husband leaves you...taking the computer with him and you
     call him crying, and beg him to bring the computer back.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

andys8 commented:
Awesome, thanks a lot :)
